libraries/AP_EFI/AP_EFI_Backend.h

```diff
@@ -33,6 +33,7 @@ class AP_EFI_Backend {
 
 #if AP_SCRIPTING_ENABLED
     virtual bool handle_scripting(const EFI_State &efi_state) { return false; }
+    virtual EFI_State get_state() { return internal_state; }
```

Comment:
I agree this is sensible. But now I'm wondering if more of the EFI scripting support should be moved to the frontend class instead of accessing the frontend through the backend

Reply:
I see now that the EFI frontend has a get_state function, I don't know how I missed that before. But it returns the state through the parameter. Is that possible through lua? Alternatively, should the frontend get_state just return an EFI_State, I don't think it will affect the stack use, but that does involve a potentially breaking change so I don't think that's the way to go

Reply:
Yeah, thanks makes sense. You would use the ```'Ref``` modifier. EG:
```
singleton AP_EFI method get_state void EFI_State'Ref
```

Reply:
What do you think of it now?